visualize.py

```diff
@@ -0,0 +1,263 @@
+    ("iso-8859-7", 16, "el"),
+    ("iso-8859-8", 16, "he"),
+    ("iso-8859-10", 16, "no"),
+    ("iso-8859-13", 16, "et"),
+    ("iso-8859-14", 16, "ga"),
+    ("iso-8859-15", 16, "de"),
+    ("iso-8859-16", 16, "pl"),
+    ("koi8-r", 16, "ru"),
+    ("koi8-u", 16, "uk"),
+    ("macintosh", 16, "de"),
+    ("windows-874", 16, "th"),
+    ("windows-1250", 16, "pl"),
+    ("windows-1251", 16, "ru"),
+    ("windows-1252", 16, "de"),
+    ("windows-1253", 16, "el"),
+    ("windows-1254", 16, "tr"),
+    ("windows-1255", 16, "he"),
+    ("windows-1256", 16, "ar"),
+    ("windows-1257", 16, "et"),
+    ("windows-1258", 16, "vi"),
+    ("x-mac-cyrillic", 16, "ru"),
+    ("jis0208", 94, "ja"),
+    ("jis0212", 94, "ja"),
+    ("euc-kr", 190, "ko"),
+    ("gb18030", 190, "zh-cn"),
+    ("big5", 157, "zh-tw"),
+]
+
+# Lead offset one, lead offset two,
+# trail offset one, trail offset two,
+byte_rules = {
+    "jis0208": (0xA1, None, 0xA1, None),
+    "jis0212": (0xA1, None, 0xA1, None),
+    "shift_jis": (0x81, 0xC1, 0x40, 0x41),
+    "euc-kr": (0x81, None, 0x41, None),
+    "gb18030": (0x81, None, 0x40, 0x41),
+    "big5": (0x81, None, 0x40, 0x62),
+}
+
+big5_prefer_last = [
+  0x2550,
+  0x255E,
+  0x2561,
+  0x256A,
+  0x5341,
+  0x5345,
+]
+
+
+def classify(code_point):
+    if code_point < 0x80:
+        raise Error()
+    if code_point < 0x800:
+        return "mid"
+    if code_point > 0xFFFF:
+        return "astral"
+    if code_point >= 0xE000 and code_point <= 0xF8FF:
+        return "pua"
+    return "upper"
+
+
+def check_compatibility(code_point):
+    if code_point >= 0xF900 and code_point <= 0xFAFF:
+        return " compatibility"
+    if code_point >= 0x3400 and code_point <= 0x4DB5:
+        return " ext"
+    return ""
+
+
+def aria(code_point, contiguous, duplicate):
+    label = None
+    classification = classify(code_point)
+    if classification == "mid":
+        label = "Two bytes"
+    elif classification == "upper":
+        label = "Three bytes"
+    elif classification == "pua":
+        label = "Three bytes, Private Use"
+    elif classification == "astral":
+        label = "Four bytes"
+    block = check_compatibility(code_point)
+    if block == " compatibility":
+        label += ", Compatibility Ideograph"
+    elif block == " ext":
+        label += "Ideograph Extension A"
+    elif code_point >= 0x4E00 and code_point <= 0x9FD5:
+        label += ", Unified Ideograph"
+    elif code_point >= 0xAC00 and code_point <= 0xD7AF:
+        label += ", Hangul"
+    if contiguous:
+        label += ", contiguous"
+    if duplicate:
+        label += ", duplicate"
+    return label
+
+
+def format_code_point(code_point):
+    if code_point >= 0x80 and code_point < 0xA0:
+        # HTML prohibits C1 controls
+        # TODO draw some fancy SVG hex inside the square
+        return "<svg><rect x=1 y=1 width=14 height=14 stroke=black stroke-width=2 fill=none /></svg>"
+    as_str = unichr(code_point)
```

Comment:
Python's sad Unicode support strikes again. How did you get your copy of Python? Apple?

The script works on Python 2.7.12 shipped on Ubuntu 16.04. Debian and, therefore, Ubuntu ships wide Python. It seems that Travis also runs Ubuntu, so it should be OK there.

@annevk, Is making the script work on e.g. Apple-shipped Python a blocker for merging?